src/traces/bar/cross_trace_calc.js

```diff
@@ -356,7 +356,7 @@ function applyAttributes(sieve) {
 
         if(isArrayOrTypedArray(offset)) {
             // if offset is an array, then clone it into t.poffset.
-            newPoffset = offset.slice(0, calcTrace.length);
+            newPoffset = Array.prototype.slice.call(offset, 0, calcTrace.length);
```

Comment:
This here converts a typed array to a regular array. Note that `slice` for typed array works, but below e.g.

https://github.com/plotly/plotly.js/blob/0a310c99455bf7389f4bb891392c806b59d93d47/src/traces/bar/cross_trace_calc.js#L432-L435

we expect computed offsets and widths to be within regular arrays.

Reply:
Is it more complicated than turning those `Array.isArray` calls into `isArrayOrTypedArray`? The way you have it here seems OK, because bars never have *that* many elements, but it seems a shame to convert back to regular arrays early if it's not too hard to support typed arrays.

Reply:
> Is it more complicated than turning those `Array.isArray` calls into `isArrayOrTypedArray`?

Yes because the rest of `crossTraceCalc` assumes you have arrays you can `.push`ed into e.g.:

https://github.com/plotly/plotly.js/blob/aea6b8d796d367c3930da9753a8b9151e10cd5d8/src/traces/bar/cross_trace_calc.js#L405-L407

which can't be done using typed arrays.

Reply:
Ah OK, makes sense. We can revisit if/when we get rid of `calcdata`, before that it doesn't make much difference anyway.